src/parquet/util/memory.h

```diff
@@ -31,35 +31,38 @@
 #include "arrow/io/memory.h"
 #include "arrow/memory_pool.h"
 #include "arrow/status.h"
+#include "arrow/util/compression.h"
 
 #include "parquet/exception.h"
+#include "parquet/types.h"
 #include "parquet/util/macros.h"
 #include "parquet/util/visibility.h"
 
-#define PARQUET_CATCH_NOT_OK(s)                    \
-  try {                                            \
-    (s);                                           \
-  } catch (const ::parquet::ParquetException& e) { \
-    return ::arrow::Status::IOError(e.what());     \
-  }
-
-#define PARQUET_IGNORE_NOT_OK(s) \
-  try {                          \
-    (s);                         \
-  } catch (const ::parquet::ParquetException& e) { UNUSED(e); }
-
-#define PARQUET_THROW_NOT_OK(s)                     \
-  do {                                              \
-    ::arrow::Status _s = (s);                       \
-    if (!_s.ok()) {                                 \
-      std::stringstream ss;                         \
-      ss << "Arrow error: " << _s.ToString();       \
-      ::parquet::ParquetException::Throw(ss.str()); \
-    }                                               \
-  } while (0);
-
 namespace parquet {
 
+static inline std::unique_ptr<::arrow::Codec> GetArrowCodec(Compression::type codec) {
```

Comment:
How about change to `GetCodecFromArrow` and reuse use codec_type?

ArrowCodec seems a bit weird..